flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/TwoPhaseCommitSinkFunction.java

```diff
@@ -255,7 +255,6 @@ public final void notifyCheckpointComplete(long checkpointId) throws Exception {
 		//
```

Comment:
One thing aboved comment should be adapted:

>  		// ==> There should never be a case where we have no pending transaction here

Btw @StefanRRichter are you sure that subsuming can happen as in this test that you added? Basing on this large comment (@StephanEwen written it for pravega connector), subsuming of notification should happen on JobManager and such subsumed notification shouldn't reach the task.

Reply:
That is a very good point and IIRC, that is the case. Hmm, I see that there is always an element added in the snapshot method, so one should expect that there is always one in the notify part because no other method is ever removing from the map and method calls should also be mutually exclusive because they both hold the checkpointing lock. What else is left?

Reply:
That is a very good point and IIRC, that is the case. Hmm, I see that there is always an element added in the snapshot method, so one should expect that there is always one in the notify part because no other method is ever removing from the map and method calls should also be mutually exclusive because they both hold the checkpointing lock. What else is left?

Reply:
Maybe you can ask the reporter of the bug to provide debug or at least info logs?

Reply:
Hi @StefanRRichter @pnowojski 
I found that this test would happen if the previous one is a savepoint, which will not be subsumed by JobManager. Please correct me if I misunderstood.

 I think we still need to release this constraint. What do you think?

Reply:
Thanks for pointing this out @tony810430. I'm re-opening this PR.